spring-rabbit/src/main/java/org/springframework/amqp/rabbit/listener/SimpleMessageListenerContainer.java

```diff
@@ -996,8 +1011,14 @@ private boolean doReceiveAndExecute(BlockingQueueConsumer consumer) throws Excep
 
 		List<Message> messages = null;
 		long deliveryTag = 0;
+		long startTime = System.currentTimeMillis();
 
 		for (int i = 0; i < this.batchSize; i++) {
+			boolean batchTimedOut = (System.currentTimeMillis() - startTime) > this.batchReceiveTimeout;
+			if (batchTimedOut) {
+				logger.trace("Timed out for gather batch messages.");
```

Comment:
It might be useful to mention also how many messages are gathered for this batch during this interval.